pymc4/coroutine_model.py

```diff
@@ -61,21 +61,23 @@ class ModelTemplate:
     """Model Template -- generative model with metadata.
 
     ModelTemplate is a callable object that represents a generative process. A generative process samples
-    from prior distributions and allows them to interact in arbitrarily-complex, user-defined ways.
+    from prior distributions and allows them to interact in arbitrarily complex, user-defined ways.
 
     Parameters
     ----------
     template : callable
-        generative process, that accepts any arguments as conditioners and returns realizations if any.
+        Generative process, that accepts any arguments as conditioners and returns realizations if any.
+    name : str, optional
+        Name for the model, inferred from defaults if ``None``.
     keep_auxiliary : bool
-        generative process may require some auxiliary variables to be created, but they are probably will not be used
-        anywhere else. In that case it is useful to tell PyMC4 engine that we can get rid of auxiliary variables
-        as long as they are not needed any more.
+        The generative process (``template``) may require some auxiliary variables to be created,
+        but they probably will not be used anywhere else. In that case it is useful to tell the
+        PyMC4 engine that it can get rid of them.
     keep_return : bool
-        the return value of the model will be recorded
+        The return value of the model will be recorded.
```

Comment:
Can you say where/how the return value will be recorded?

Also, maybe clarify what is meant by "return value", since this is a class not a function, and we have said that we use `yield`, which makes a stream of values instead of a return value.